### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by marius-se**
This PR focusses on the first step of the registration step. At the core it adds a new function `beginRegistration` and renames a couple of types to align with the WebAuthn specs.

### CODE REVIEW

**Comment by 0xTim** [comment]
@swift-server-bot add to allowlist

File: Tests/WebAuthnTests/HelpersTests.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        XCTAssertNotEqual(challenge1, challenge2)
+    }
+
+    func testBase64URLEncodeReturnsCorrectString() {
+        let input: [UInt8] = [1, 0, 1, 0, 1, 1]
+        let expectedBase64 = Data(bytes: input, count: input.count).base64EncodedString()
+
+        let base64URLEncoded = input.base64URLEncodedString()
+        let base64Encoded = base64URLEncoded.replacingOccurrences(of: "-", with: "+")
+            .replacingOccurrences(of: "_", with: "/")
```
Comment (0xTim) [inline Tests/WebAuthnTests/HelpersTests.swift:19]: This expected value can be hardcoded right?

**Review by 0xTim — APPROVED** [review]
Looks great! Couple of minor comments

File: Tests/WebAuthnTests/HelpersTests.swift (hunk 1 above)
Comment (marius-se) [reply Tests/WebAuthnTests/HelpersTests.swift]: Yep

File: Sources/WebAuthn/PublicKeyCredentialParameters.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                in: container,
+                debugDescription: "Currently only public-key is supported."
+            )
+        }
+
+        let algorithm = try container.decode(String.self, forKey: .alg)
+        switch algorithm {
+        case "AlgES256": self = .algES256
+        default:
+            throw DecodingError.dataCorruptedError(
```
[the remark this answers is not in the document]
Comment (marius-se) [reply Sources/WebAuthn/PublicKeyCredentialParameters.swift:24]: Is there a more specific error? Afaik that's the Foundation default error also used by `Codable`, etc. ...?

[the remark this answers is not in the document]
Comment (marius-se) [reply Sources/WebAuthn/PublicKeyCredentialParameters.swift:24]: decoding logic will change with second PR - not relevant anymore - closing.

**Pull request merged**